Client/Frontend/Browser/OpenSearch.swift

```diff
@@ -19,35 +22,43 @@ class OpenSearchEngine {
     let shortName: String
     let description: String?
     let image: UIImage?
-    private let urls: [OpenSearchURL]
+    private let searchURL: OpenSearchURL
+    private let suggestURL: OpenSearchURL?
 
-    private init(shortName: String, description: String?, urls: [OpenSearchURL], image: UIImage?) {
+    private init(shortName: String, description: String?, image: UIImage?, searchURL: OpenSearchURL, suggestURL: OpenSearchURL?) {
         self.shortName = shortName
         self.description = description
-        self.urls = urls
         self.image = image
+        self.searchURL = searchURL
+        self.suggestURL = suggestURL
     }
 
     /**
      * Returns the search URL for the given query.
-     *
-     * The <Url> element with the "text/html" type is used for the template.
      */
-    func urlForQuery(query: String) -> NSURL? {
-        for url in urls {
-            if url.type == "text/html" {
-                let escapedQuery = query.stringByAddingPercentEncodingWithAllowedCharacters(NSCharacterSet.URLQueryAllowedCharacterSet())
-                if escapedQuery == nil {
-                    return nil
-                }
+    func searchURLForQuery(query: String) -> NSURL? {
+        return getURLFromTemplate(searchURL, query: query)
+    }
 
-                let urlString = url.template.stringByReplacingOccurrencesOfString("{searchTerms}", withString: escapedQuery!, options: NSStringCompareOptions.LiteralSearch, range: nil)
-                return NSURL(string: urlString)
-            }
+    /**
+     * Returns the search suggestion URL for the given query.
+     */
+    func suggestURLForQuery(query: String) -> NSURL? {
+        if suggestURL == nil {
+            return nil
+        }
+
+        return getURLFromTemplate(suggestURL!, query: query)
+    }
+
+    private func getURLFromTemplate(openSearchURL: OpenSearchURL, query: String) -> NSURL? {
+        let escapedQuery = query.stringByAddingPercentEncodingWithAllowedCharacters(NSCharacterSet.URLQueryAllowedCharacterSet())
```

Comment:
Nit: this is a good candidate for `if let = query.syringBy...() { }`. Saves a line.